### ISSUE TITLE
Harder to convergence without 'padding_idx=0' in 'nn.Embedding()' 

### ISSUE THREAD

**Issue by OlaWod** [NVIDIA/tacotron2#484]
https://github.com/NVIDIA/tacotron2/blob/185cd24e046cc1304b4f8e564734d2498c6e2e6f/model.py#L464
https://github.com/NVIDIA/tacotron2/blob/185cd24e046cc1304b4f8e564734d2498c6e2e6f/model.py#L465

I found that training with `self.embedding = nn.Embedding(hparams.n_symbols, hparams.symbols_embedding_dim)` will result in alignment like this below:
![IMG_0687(20210509-133132)](https://user-images.githubusercontent.com/32514635/117561815-ac920600-b0cc-11eb-9c7e-15f12b48d639.PNG)

And there are similar issues too:
https://github.com/NVIDIA/tacotron2/issues/461#issue-837032398
https://github.com/NVIDIA/tacotron2/issues/135#issuecomment-466423674
https://github.com/NVIDIA/tacotron2/issues/63#issuecomment-409179420

But if add 'padding_idx=0', i.e. `self.embedding = nn.Embedding(hparams.n_symbols, hparams.symbols_embedding_dim, padding_idx=0)`, the alignment begin converge with only 8k steps and the training speed is faster due to more sparse matrix.
![IMG_0698(20210509-133042)](https://user-images.githubusercontent.com/32514635/117561813-ab60d900-b0cc-11eb-9183-e0ba0f555faa.JPEG)

I train with LJSpeech dataset, batch_size=32 with my code https://github.com/OlaWod/my-tacotron2 modified from this repo. Text is converted to phoneme. The model part in the code is almost the same with this repo.

-----

In conclusion:

+ Suggest adding 'padding_idx' in nn.Embedding(), at least it can speed up training due to more sparse matrix.
+ Wonder why alignment is horizontal line without 'padding_idx'? Is it because there're something wrong in my modified version code? But there are other person whose alignment looks like this too, and they run with this repo.

Thanks.

**Comment by OlaWod** [comment]
https://github.com/NVIDIA/tacotron2/blob/185cd24e046cc1304b4f8e564734d2498c6e2e6f/model.py#L468
https://github.com/NVIDIA/tacotron2/blob/185cd24e046cc1304b4f8e564734d2498c6e2e6f/layers.py#L13
https://github.com/NVIDIA/tacotron2/blob/185cd24e046cc1304b4f8e564734d2498c6e2e6f/layers.py#L34
Found performance improve a lot in both situations if add weight uniform.
With padding_idx alignment diagonal emerge at step 5000, 
without padding_idx alignment diagonal emerge at step 8500.
